Source/Core/DolphinQt/GameList/GameTracker.cpp

```diff
@@ -1,240 +1,86 @@
-	}
-	if (SConfig::GetInstance().m_ListWii || SConfig::GetInstance().m_ListGC)
-	{
-		exts.push_back(".iso");
-		exts.push_back(".ciso");
-		exts.push_back(".wbfs");
-	}
-	if (SConfig::GetInstance().m_ListWad)
-		exts.push_back(".wad");
-	if (SConfig::GetInstance().m_ListElfDol)
-	{
-		exts.push_back(".dol");
-		exts.push_back(".elf");
-	}
-
-	auto rFilenames = DoFileSearch(exts, SConfig::GetInstance().m_ISOFolder, SConfig::GetInstance().m_RecursiveISOFolder);
-	QList<GameFile*> newItems;
-	QStringList allItems;
-
-	if (!rFilenames.empty())
+	QDirIterator it(dir, m_filters);
+	while (it.hasNext())
 	{
-		for (u32 i = 0; i < rFilenames.size(); i++)
+		QString path = QFileInfo(it.next()).canonicalFilePath();
+		if (!m_tracked_files.contains(path))
 		{
-			std::string FileName;
-			SplitPath(rFilenames[i], nullptr, &FileName, nullptr);
-			QString NameAndPath = QString::fromStdString(rFilenames[i]);
-			allItems.append(NameAndPath);
-
-			if (m_games.keys().contains(NameAndPath))
-				continue;
-
-			GameFile* obj = new GameFile(rFilenames[i]);
-			if (obj->IsValid())
+			if (addPath(path))
 			{
-				bool list = true;
-
-				switch (obj->GetCountry())
-				{
-					case DiscIO::IVolume::COUNTRY_AUSTRALIA:
-						if (!SConfig::GetInstance().m_ListAustralia)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_EUROPE:
-						if (!SConfig::GetInstance().m_ListPal)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_FRANCE:
-						if (!SConfig::GetInstance().m_ListFrance)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_GERMANY:
-						if (!SConfig::GetInstance().m_ListGermany)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_ITALY:
-						if (!SConfig::GetInstance().m_ListItaly)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_JAPAN:
-						if (!SConfig::GetInstance().m_ListJap)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_KOREA:
-						if (!SConfig::GetInstance().m_ListKorea)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_NETHERLANDS:
-						if (!SConfig::GetInstance().m_ListNetherlands)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_RUSSIA:
-						if (!SConfig::GetInstance().m_ListRussia)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_SPAIN:
-						if (!SConfig::GetInstance().m_ListSpain)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_TAIWAN:
-						if (!SConfig::GetInstance().m_ListTaiwan)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_USA:
-						if (!SConfig::GetInstance().m_ListUsa)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_WORLD:
-						if (!SConfig::GetInstance().m_ListWorld)
-							list = false;
-						break;
-					case DiscIO::IVolume::COUNTRY_UNKNOWN:
-					default:
-						if (!SConfig::GetInstance().m_ListUnknown)
-							list = false;
-						break;
```

Comment:
This is an important feature -- hiding specific games based on country. It may "look atrocious" to you, but unless you have a better way of doing it, it stays.